source/guides/recipes/autoformatting.md

```diff
@@ -0,0 +1,44 @@
+(autoformatting)=
+# Autoformatting
+
+[`nixfmt`](https://github.com/NixOS/nixfmt) is the official Nix autoformatter.
+
+It is not yet wired up to any official tooling. Subscribe to [PR
+11252 to NixOS/nix](https://github.com/NixOS/nix/pull/11252) for updates on
+that effort.
+
+The easiest way to leverage `nixfmt` in your project is with the `nixfmt-tree`
+wrapper. Here's how:
+
+- For `nix fmt` to format all Nix files, add this to the `flake.nix` outputs:
+
+  ```nix
+  formatter.''${system} = nixpkgs.legacyPackages.''${system}.nixfmt-tree;
+  ```
+
+- The same can be done more efficiently with the `treefmt` command,
```

Comment:
Please show exactly one way of doing things, and since official documentation is committed to stable features (and you already point out that it's more efficient) let's use the one where you add the package to your shell environment.

Reply:
> Please show exactly one way of doing things

I'm not sure I agree we should limit ourselves to only documenting a single approach to things.

`nix fmt` and `nix-shell --run treefmt` both have their own pros and cons, I'd expect users would want to be aware of each.

Reply:
Realty is, we have to balance cognitive load against factual completeness. This place is about managing cognitive load, reference documentation is for completeness. I'd appreciate  discussion in reference documentation and issue threads on the costs and benefits of various approaches, but the motto of nix.dev is "getting things done" -- not "agonizing over the options". We're free to agonize in these behind the scenes exchanges, but we must not spill that on beginners.

Reply:
Tbh I'm not sure if `nix fmt` has a benefit other than it being a shorter command, but we should recommend `direnv` anyways, in which case `treefmt` has the exact same length. I'd be fine with dropping `nix fmt`, though open about other benefits I'm missing!